src/hotspot/share/prims/whitebox.cpp

```diff
@@ -840,11 +859,32 @@ WB_ENTRY(jboolean, WB_IsMethodCompilable(JNIEnv* env, jobject o, jobject method,
   CHECK_JNI_EXCEPTION_(env, JNI_FALSE);
   MutexLocker mu(Compile_lock);
   methodHandle mh(THREAD, Method::checked_resolve_jmethod_id(jmid));
-  if (is_osr) {
-    return CompilationPolicy::can_be_osr_compiled(mh, comp_level);
-  } else {
-    return CompilationPolicy::can_be_compiled(mh, comp_level);
+
+  // The ExcludeOption directive is evaluated lazily upon compilation attempt. If a method was not tried to be compiled by
+  // a compiler, yet, the method object is not set to be not compilable by that compiler. Thus, evaluate the compiler directive
+  // to exclude a compilation of 'method'.
+  if (comp_level == CompLevel_any) {
+    // Both compilers could have ExcludeOption set. Check all combinations.
+    bool excluded_c1 = is_excluded_for_compiler(CompileBroker::compiler1(), mh);
+    bool excluded_c2 = is_excluded_for_compiler(CompileBroker::compiler2(), mh);
```

Comment:
May be use next instead as we do in `WhiteBox::compile_method` at line #992:
```AbstractCompiler *comp = CompileBroker::compiler(comp_level);```

Reply:
The problem is that `CompileBroker::compiler()` returns `NULL` for `CompLevel_any`. And even if it returned one compiler, I also need to check the other one to decide if the method is completly non-compilable. That's why I added this additional logic for `CompLevel_any`.

Reply:
I thought `exclude` command does not specify which compilation level (and corresponding compiler) is disabled - it disables all compilations.
But may be it is not true for directives. @neliasso, please, correct me if I am wrong.

Reply:
> I thought `exclude` command does not specify which compilation level (and corresponding compiler) is disabled - it disables all compilations.
> But may be it is not true for directives. @neliasso, please, correct me if I am wrong.

You can set Exclude differently for c1 and c2. That's sometimes very handy when creating more complex combinations of directives.

Reply:
Changes good then.